internal/pkg/service/stream/storage/statistics/statistics.go

```diff
@@ -67,30 +67,40 @@ type Aggregated struct {
 	Total Value
 }
 
-// Add returns a sum of the value and the given second value.
-// If only one of the values is a reset value then it returns a non-reset value using subtraction.
-// Note that the value can't be negative, in case of an underflow the fields will be 0.
-func (v Value) Add(v2 Value) Value {
-	r := v
-	if (r.ResetAt == nil) != (v2.ResetAt == nil) {
-		if r.ResetAt != nil {
-			return v2.Add(r)
-		}
-		r = r.sub(v2)
+func (v Value) With(v2 Value) Value {
+	if v.ResetAt != nil {
```

Comment:
`Add` -> `With` bacause it is immutable, so it is less confusing.

---------

- Added `add/sub` private methods.
- `With` method cannot be called on a reset value (to simplify code), bude can be called WITH reset value.
  - Because the `sum` container is never a reset value.